### ISSUE TITLE
[unused-private-member] false positive for property()

### ISSUE THREAD

**Issue by stanislavlevin** [PyCQA/pylint#4756]
### Bug description

Given a file `a.py`:

```python
# pylint: disable=missing-docstring,too-few-public-methods
class Prop:
    def __get_prop(self):
        pass

    def __set_prop(self, value):
        pass

    def __del_prop(self):
        pass

    prop = property(__get_prop, __set_prop, __del_prop)
```

### (Configuration)

Using the following configuration:
```ini
```

### Command used

Result of `pylint a.py`:
```
************* Module a
a.py:3:4: W0238: Unused private member `Prop.__get_prop(self)` (unused-private-member)
a.py:6:4: W0238: Unused private member `Prop.__set_prop(self, value)` (unused-private-member)
a.py:9:4: W0238: Unused private member `Prop.__del_prop(self)` (unused-private-member)

-----------------------------------
Your code has been rated at 6.25/10
```

### (Expected behavior)

There should be no errors.

### (Version affected)

```shell
pylint 2.9.0+ (including today's main).
```

### (OS / Environment)

ALTLinux

**Comment by yushao2** [comment]
looking into this, thanks for the report!

**Comment by spaghEddieDoyle** [comment]
@yushao2 I have had a similar issue, and it seems to me it can be generalized to variables referenced in the class body without dot-methods. 
```python
# test.py 
from functools import partialmethod
class Foo:
    def __bar(self, x):
        print(x)
    fizz = partialmethod(__bar, 'fizz')
foo = Foo()
foo.fizz()
```

```shell
$ python test.py
fizz
```
```shell
$ pylint --version
pylint 2.10.2
astroid 2.7.2
Python 3.7.5 (default, Aug 13 2020, 09:55:33) 
[Clang 11.0.3 (clang-1103.0.32.62)]
```

**Comment by stanislavlevin** [comment]
great!
the fix verified, thank you!
